File: benchmarks/workload.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

static uint64_t mix_value(uint64_t value) {
    value ^= value >> 12;
    value ^= value << 25;
    value ^= value >> 27;
    return value * UINT64_C(2685821657736338717);
}

int main(int argc, char **argv) {
    uint64_t rounds = UINT64_C(25000000);
    if (argc > 1) rounds = strtoull(argv[1], NULL, 10);
    uint64_t state = UINT64_C(88172645463325252);
    uint64_t checksum = 0;
    for (uint64_t index = 0; index < rounds; ++index) {
        state = mix_value(state + index);
        checksum ^= state;
    }
    printf("%llu\n", (unsigned long long)checksum);
    return 0;
}

File: benchmarks/workload.rs
use std::env;

#[inline]
fn mix_value(mut value: u64) -> u64 {
    value ^= value >> 12;
    value ^= value << 25;
    value ^= value >> 27;
    value.wrapping_mul(2_685_821_657_736_338_717)
}

fn main() {
    let rounds = env::args().nth(1).and_then(|v| v.parse().ok()).unwrap_or(25_000_000_u64);
    let mut state = 88_172_645_463_325_252_u64;
    let mut checksum = 0_u64;
    for index in 0..rounds {
        state = mix_value(state.wrapping_add(index));
        checksum ^= state;
    }
    println!("{checksum}");
}
